gapir/cc/interpreter.cpp

```diff
@@ -337,6 +341,49 @@ Interpreter::Result Interpreter::resource(uint32_t opcode) {
   return this->call(Interpreter::RESOURCE_FUNCTION_ID);
 }
 
+Interpreter::Result Interpreter::inlineResource(uint32_t opcode) {
+  unsigned int numValuePatchUps = extract6bitData(opcode);
+  unsigned int dataSize = extract20bitData(opcode);
+
+  void* destination = mStack.pop<void*>();
+  memcpy(destination, &mInstructions[mCurrentInstruction + 1],
+         (size_t)dataSize);
+
+  auto roundedDataSize = (dataSize / 4) + (dataSize % 4 != 0 ? 1 : 0);
+
+  for (unsigned int i = 0; i < numValuePatchUps; ++i) {
+    uint32_t destination =
+        mInstructions[mCurrentInstruction + 1 + roundedDataSize + i * 2];
+    uint32_t value =
+        mInstructions[mCurrentInstruction + 1 + roundedDataSize + i * 2 + 1];
+
+    *((void**)mMemoryManager->volatileToAbsolute(destination)) =
```

Comment:
This makes some assumptions about the size of the patchups?
What happens if the patched value is not sizeof(void*) (This is quite normal in Vulkan for example), and DEFINITELY normal in GL.

Reply:
The "patch ups" offered by this instruction are only applicable to void* sized data. Maybe this should be commented to a greater degree. Unless I misunderstand the memory manager, the way we're using this in builder.go should not insert "patch ups" for things that aren't pointer sized? If there is a problem we should address it there. I did not want to reimplement the entire type system inside this instruction.

Reply:
So, looking more carefully it looks like we are only patching pointers?
Is there a particular reason for that? Because in Vulkan at least, there are a TON of handles that also get patched, that this could benefit.

Reply:
Will address in follow up commit that extends the functionality.